src/databricks/azext_databricks/_validators.py

```diff
@@ -37,3 +38,14 @@ def validate_workspace_values(cmd, namespace):
             namespace='Microsoft.Network',
             type='virtualNetworks',
             name=namespace.custom_virtual_network_id)
+
+
+def validate_encryption_values(namespace):
+    from knack.util import CLIError
+    if namespace.encryption_key_source:
+        if namespace.encryption_key_source == 'Default' and any(v is not None for v in [namespace.encryption_key_name, namespace.encryption_key_version, namespace.encryption_key_vault]):
+            raise CLIError('--key-name, --key-version, --key-vault should not be provided when --key-source is Default')
+        if namespace.encryption_key_source == 'Microsoft.Keyvault' and any(v is None for v in [namespace.encryption_key_name, namespace.encryption_key_version, namespace.encryption_key_vault]):
+            raise CLIError('--key-name, --key-version, --key-vault are required when --key-source is Microsoft.Keyvault')
+    elif any(v is not None for v in [namespace.encryption_key_name, namespace.encryption_key_version, namespace.encryption_key_vault]):

```

Comment:
if key_source is None, why still need check key_name, version etc?

Reply:
To consider the case when key_source is forgotten but other key options are provided.

Reply:
Hi @fengzhou-msft , same concern with Catherine here: what if users don't specify any of these 4 parameters? It seems this validator would be passed.

Reply:
It is a valid case, the key properties are not updated. They are not required in this update command.